### ISSUE TITLE
Error in creating docker container from webR repository 

### ISSUE THREAD

**Issue by SugarRayLua** [georgestagg/webR#21]
Hello.

I’m a novice programmer very interested in what you created with webR and attempted today to compile my own wasm files from your repository source. As I understood the instructions, I was supposed to create a docker image using the Dockerfile in the webR repository then create a docker container using that docker image in which I would clone in the repository into a new director in that container then got to the top of that directory and “./configure && make”. However, I tried that process multiple times and was able to create a docker image fine but not an appropriate container image. Each time I would go to use the new container I created I found that the container already stopped itself due to a make error listen in the docker container log (“ Make: *** No rule to make target ‘webr’. Stop.)

I’m using Docker on on a x64 iMac running OSX Monterey. Here is an example of one of my attempts to create a usable docker container to put webR in:

-iMac ~ % docker build https://github.com/georgestagg/webR.git#main
[+] Building 2.1s (9/9) FINISHED                                                                                                                             
 => [internal] load git source https://github.com/georgestagg/webR.git#main                                                                             0.8s
 => [internal] load metadata for docker.io/library/ubuntu:20.04                                                                                         1.2s
 => [auth] library/ubuntu:pull token for registry-1.docker.io                                                                                           0.0s 
 => [1/5] FROM docker.io/library/ubuntu:20.04@sha256:47f14534bda344d9fe6ffd6effb95eefe579f4be0d508b7445cf77f61a0e5724                                   0.0s 
 => CACHED [2/5] RUN echo "deb http://archive.ubuntu.com/ubuntu/ trusty main restricted universe" >> /etc/apt/sources.list &&     echo "deb http://sec  0.0s
 => CACHED [3/5] WORKDIR /app                                                                                                                           0.0s
 => CACHED [4/5] RUN mkdir -p /webr-tools &&     cd /webr-tools &&     git clone --depth=1 https://github.com/emscripten-core/emsdk.git &&     cd emsd  0.0s
 => CACHED [5/5] RUN ln -s /usr/bin/gfortran-4.6 /usr/bin/gfortran                                                                                      0.0s
 => exporting to image                                                                                                                                  0.0s
 => => exporting layers                                                                                                                                 0.0s
 => => writing image sha256:ac280cc7722234b9b339e88fa3a954aaea77ad58b9698da66975546b3bca2ce0    

iMac ~ % docker create ac280cc77222

(However, every time I then attempted to enter the above container to try and clone in the repository, it would give me the error that the container I created from the above command had stopped running. I would restart the container but the container would stop itself again; when I went to the log file for the docker container it displayed the following:)    

Make: *** No rule to make target ‘webr’. Stop.

Am I doing something wrong in this process? 

I appreciate any advice you could give.

Thanks! :-)

**Comment by georgestagg** [comment]
The instructions are a bit out of date (sorry!) but by using flang rather than dragonegg one can build webR without the docker image. Probably on macOS this is the best way to go.

To do that, don't bother building the Docker image or starting docker, just clone the repo, cd to it, then run `./configure` and `make`. The first time you do this it will take a very long time as it builds everything. At the end of the process you should have a copy of webR in the `./dist` directory.

**Comment by SugarRayLua** [comment]
Thanks, I will try that. I’m mostly away from my iMac during the week (hence my interest to have R run natively on my iPad) so may be next weekend until I’m able to fully test.

Sincerely,
Mike

**Comment by SugarRayLua** [comment]
@georgestagg , do I have to have emscripten also installed already on my Mac? I cloned the repo at the end of the weekend and let it ./configure and make but ended up with two make errors (127 and 2). If I needed to have emscripten installed, I’ll installed emscripten next weekend when I’m back at my Mac and try those same instructions again. Thanks, Mike

**Comment by georgestagg** [comment]
Yes, Emscripten is required first. We should add a check for that in the configure script...

You can install Emscripten using the `emsdk` instructions here: https://emscripten.org/docs/getting_started/downloads.html

**Comment by SugarRayLua** [comment]
Thanks! Will try it out this weekend. Sincerely, Mike
